overlord/snapshotstate/backend/backend.go

```diff
@@ -369,3 +374,158 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 
 	return nil
 }
+
+type exportMetadata struct {
+	Format int       `json:"format"`
+	Date   time.Time `json:"date"`
+	Files  []string  `json:"files"`
+}
+
+type SnapshotExport struct {
+	// open snapshot files
+	snapshotFiles []*os.File
+
+	// remember setID mostly for nicer errors
+	setID uint64
+
+	// cached size, needs to be calculated with CalculateSize
+	size int64
+}
+
+// NewSnapshotExport will return a SnapshotExport structure. It must be
+// Close()ed after use to avoid leaking file descriptors.
+func NewSnapshotExport(ctx context.Context, setID uint64) (se *SnapshotExport, err error) {
+	var snapshotFiles []*os.File
```

Comment:
Are we expected to ever have more than one snapshot matching `setID`? I'm worried about keeping unbound number of file descriptors open. Could this be reduced to just one `snapshotFile`?

Reply:
Unfortunately each snapshot id can have multiple zip files.